cli/commands/preview/tests/create.test.ts

```diff
@@ -1,82 +1,325 @@
+
+		await program.parseAsync( [ 'node', 'test', 'go', mockFolder ] );
+
+		expect( mockLogger.reportError ).toHaveBeenCalled();
+		expect( mockErrorData ).toHaveProperty( 'message', errorMessage );
+		expect( mockErrorData ).toHaveProperty( 'action', 'upload' );
+		expect( waitForSiteReady ).not.toHaveBeenCalled();
 	} );
 
-	it( 'should handle errors gracefully', async () => {
-		const mockError = new Error( 'Test error' );
-		( fs.createWriteStream as jest.Mock ).mockImplementation( () => {
-			throw mockError;
+	it( 'should handle site readiness errors', async () => {
+		const errorMessage = 'Failed to create preview site';
+		( waitForSiteReady as jest.Mock ).mockImplementation( () => {
+			throw new LoggerError( errorMessage, 'ready' );
 		} );
 
+		const { registerCommand } = await import( '../create' );
 		registerCommand( program );
+
+		await program.parseAsync( [ 'node', 'test', 'go', mockFolder ] );
+
+		expect( mockLogger.reportError ).toHaveBeenCalled();
+		expect( mockErrorData ).toHaveProperty( 'message', errorMessage );
+		expect( mockErrorData ).toHaveProperty( 'action', 'ready' );
+		expect( addPreviewSiteToAppdata ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should handle appdata errors', async () => {
+		const errorMessage = 'Failed to save to appdata';
+		( addPreviewSiteToAppdata as jest.Mock ).mockImplementation( () => {
+			throw new LoggerError( errorMessage, 'appdata' );
+		} );
+
+		const { registerCommand } = await import( '../create' );
+		registerCommand( program );
+
+		await program.parseAsync( [ 'node', 'test', 'go', mockFolder ] );
+
+		expect( mockLogger.reportError ).toHaveBeenCalled();
+		expect( mockErrorData ).toHaveProperty( 'message', errorMessage );
+		expect( mockErrorData ).toHaveProperty( 'action', 'appdata' );
+	} );
+
+	it( 'should always clean up archive file even on error', async () => {
+		( uploadArchive as jest.Mock ).mockImplementation( () => {
+			throw new LoggerError( 'Upload failed', 'upload' );
+		} );
+
+		const { registerCommand } = await import( '../create' );
+		registerCommand( program );
+
+		await program.parseAsync( [ 'node', 'test', 'go', mockFolder ] );
+
+		expect( cleanup ).toHaveBeenCalledWith( mockArchivePath );
+	} );
+
+	it( 'should handle unexpected errors', async () => {
+		const errorMessage = 'Unexpected error';
+		const unexpectedError = new Error( errorMessage );
+		( validateSiteFolder as jest.Mock ).mockImplementation( () => {
+			throw unexpectedError;
+		} );
+
+		const { registerCommand } = await import( '../create' );
+		registerCommand( program );
+
 		await program.parseAsync( [ 'node', 'test', 'go', mockFolder ] );
 
-		expect( mockLogger.reportError ).toHaveBeenCalledWith( mockError.message );
+		expect( mockLogger.reportError ).toHaveBeenCalled();
+		expect( mockErrorData ).toHaveProperty( 'message', errorMessage );
+		expect( mockErrorData ).toHaveProperty( 'action', 'validate' );
+		expect( createArchive ).not.toHaveBeenCalled();
 	} );
```

Comment:
Not wrong per se, but this strikes me as an excessive set of test cases. I assume this test code was written by AI, and it should be easy enough to modify with AI, too. Still, I think we can reduce this to two or three test cases and it would give us the same assurances in practice.